freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/user/handler/SetPasswordHandler.java

```diff
@@ -69,10 +59,17 @@ public void accept(Event<SetPasswordRequest> setPasswordRequestEvent) {
 
             FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStack(stack);
             if (FreeIpaCapabilities.hasSetPasswordHashSupport(freeIpaClient.getConfig())) {
-                LOGGER.info("IPA has password hash support, credentials information from UMS will be used.");
-                setPasswordHashFromUms(freeIpaClient, request.getUsername(), request.getUserCrn(), request.getExpirationInstant());
+                WorkloadCredential workloadCredential = umsCredentialProvider.getCredentials(request.getUserCrn(), MDCUtils.getRequestId());
+
+                LOGGER.info("IPA has password hash support. Credentials information from UMS will be used.");
+                workloadCredentialService.setWorkloadCredential(freeIpaClient, request.getUsername(), workloadCredential);
+                if (workloadCredential.getHashedPassword().isBlank()) {
```

Comment:
if you use StringUtils.isBlank here it will work even if getHashedPassword returns null